core/foundation/inc/ROOT/RLogger.hxx

```diff
@@ -16,28 +16,136 @@
 #ifndef ROOT7_RLogger
 #define ROOT7_RLogger
 
-#include <array>
+#include <atomic>
 #include <memory>
+#include <mutex>
 #include <sstream>
 #include <vector>
 
-#include "ROOT/RStringView.hxx"
-
 namespace ROOT {
 namespace Experimental {
 
 /**
  Kinds of diagnostics.
  */
 enum class ELogLevel {
-   kDebug,   ///< Debug information; only useful for developers
-   kInfo,    ///< Informational messages; used for instance for tracing
-   kWarning, ///< Warnings about likely unexpected behavior
-   kError,
-   kFatal
+   kOff = 0,
+   kFatal = 0x100,   ///< An error which causes further processing to be unreliable
+   kError = 0x200,   ///< An error
+   kWarning = 0x300, ///< Warnings about likely unexpected behavior
+   kInfo = 0x400,    ///< Informational messages; used for instance for tracing
+   kDebug = 0x500,   ///< Debug information; only useful for developers
```

Comment:
Yeah one can now decrease priority of `kError` by making it `kError + 100` or whatever, up to 255. I decided to order them this way around because of "verbosity": more verbose means higher cut-off means `kDebug` must be greater than `kError`. Let me know if you have an opinion on this, i.e. whether it's less surprising to deal with those as "priorities" / "severity" for messages, instead of "verbosity" for filtering. This enum will be used for both, so it's a question of what is more visible / relevant to users. Also @pcanal